File: repeatingCharacter.py

def repeating(a: str) -> float:
    if len(a) == 0:
        return 0
    max_ans = float('-inf')
    for i in range(len(a)):
        set = {}
        for j in range(i, len(a)):
            if a[j] in set:
                max_ans = max(max_ans, j - i)
                break
            set[a[j]] = 1
    return max_ans

if __name__ == "__main__":
    a = "takeUforward"
    print("The answer is ", repeating(a))

File: repeatingCharacter.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

int solve(string str)
{
    if (str.size() == 0)
    {
        return 0;
    }
    int maxans = INT16_MIN;
    for (int i = 0; i < str.length(); i++)
    {
        unordered_set<int> Set;
        for (int j = i; j < str.length(); j++)
        {
            if (Set.find(str[j]) != Set.end())
            {
                maxans = max(maxans, j - i);
                break;
            }
            Set.insert(str[j]);
        }
    }
    return maxans;
}

int main()
{
    string str = "takeUforward";
    cout << "The length of the longest substring is " << solve(str) << endl;
    return 0;
}